ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -512,11 +511,11 @@ private[ember] object Parser {
     }
   }
 
-  private def combineArrays[A: scala.reflect.ClassTag](a1: Array[A], a2: Array[A]): Array[A] = {
-    val buff = mutable.ArrayBuffer[A]()
-    buff.++=(a1)
-    buff.++=(a2)
-    buff.toArray
+  private[this] def appendByteArrays(a1: Array[Byte], a2: Chunk[Byte]): Array[Byte] = {
+    val res = new Array[Byte](a1.length + a2.size)
+    System.arraycopy(a1, 0, res, 0, a1.length)
+    a2.copyToArray(res, a1.length)
+    res
   }
```

Comment:
This is fantastic. @ChristopherDavenport what do you think?

Reply:
This looks good. I _think_ this code will get hit in a pass with end to end tests we have.